rotkehlchen/chain/ethereum/modules/uniswap/v3/utils.py

```diff
@@ -0,0 +1,544 @@
+    # pylint: disable=no-value-for-parameter
+    raw_address_bytes = Web3.solidityKeccak(
+        abi_types=['bytes', 'bytes'],
+        values=['0xff' + abi_encoded_2.hex(), POOL_INIT_CODE_HASH],
+    )
+    address = to_checksum_address(raw_address_bytes[12:].hex())
+    return address
+
+
+def calculate_price_range(
+        tick_lower: int,
+        tick_upper: int,
+        decimal_0: int,
+        decimal_1: int,
+) -> Tuple[FVal, FVal]:
+    """Calculates the price range for a Uniswap V3 LP position."""
+    sqrt_a = LOG_PRICE**tick_lower
+    sqrt_b = LOG_PRICE**tick_upper
+
+    sqrt_adjusted_a = sqrt_a * FVal(10**(decimal_0 - decimal_1))
+    sqrt_adjusted_b = sqrt_b * FVal(10**(decimal_0 - decimal_1))
+
+    return FVal(1 / sqrt_adjusted_b), FVal(1 / sqrt_adjusted_a)
+
+
+def compute_sqrt_values_for_amounts(
+        tick_lower: int,
+        tick_upper: int,
+        tick: int,
+) -> Tuple[FVal, FVal, FVal]:
+    """Computes the values for `sqrt`, `sqrt_a`, sqrt_b`"""
+    sqrt_a = LOG_PRICE**FVal(tick_lower / 2) * POW_96
+    sqrt_b = LOG_PRICE**FVal(tick_upper / 2) * POW_96
+    sqrt = LOG_PRICE**FVal(tick / 2) * POW_96
+    sqrt = max(min(sqrt, sqrt_b), sqrt_a)
+
+    return sqrt, sqrt_a, sqrt_b
+
+
+def calculate_amount(
+        tick_lower: int,
+        liquidity: int,
+        tick_upper: int,
+        decimals: int,
+        tick: int,
+        token_position: Literal[0, 1],
+) -> FVal:
+    """
+    Calculates the amount of a token in the Uniswap V3 LP position.
+    `token_position` can either be 0 or 1 representing the position of the token in a pair.
+    """
+    sqrt, sqrt_a, sqrt_b = compute_sqrt_values_for_amounts(
+        tick_lower=tick_lower,
+        tick_upper=tick_upper,
+        tick=tick,
+    )
+    if token_position == 0:
+        amount = (liquidity * POW_96 * (sqrt_b - sqrt) / (sqrt_b * sqrt)) / 10**decimals
+    elif token_position == 1:
+        amount = liquidity * (sqrt - sqrt_a) / POW_96 / 10**decimals
+
+    return FVal(amount)
+
+
+def calculate_total_amounts_of_tokens(
+        liquidity: int,
+        tick: int,
+        fee: Literal[100, 500, 3000, 10000],
+        decimal_0: int,
+        decimal_1: int,
+) -> Tuple[FVal, FVal]:
+    """
+    Calculates the total amount of tokens present in a liquidity pool.
+    A fee of 10000 represents 200 ticks spacing, 3000 represents 60 ticks spacing and
+    500 represents 10 ticks spacing.
+    """
+    if fee == 10000:
+        tick_a = tick - (tick % 200)
+        tick_b = tick + 200
+    elif fee == 3000:
+        tick_a = tick - (tick % 60)
+        tick_b = tick + 60
+    elif fee == 500:
+        tick_a = tick - (tick % 10)
+        tick_b = tick + 10
+    elif fee == 100:
+        tick_a = tick - (tick % 1)
+        tick_b = tick + 1
+    # This is to prevent new fee tiers from raising `referenced before initialised` error.
+    else:
+        raise AssertionError(f'Fee {fee} not supported in Uniswap V3')
```

Comment:
This is not a good idea, this will break the entire call as it will assert the minute it sees a pool with unhandled fee and kill the entire call.

you should not assert on anything that can happen. Assertions are only for things internal to our development, things that should never happen.

The proper way to handle this is just skip it and log error so user can report it to us